File: javascript/binary-search/binary-search.js
export const find = (array, value) => {

    let low = 0;
    let high = array.length;

    while (low <= high) {
        let middle = Math.floor((low + high) / 2);

        if (array[middle] === value) {
            return middle;
        }

        if (array[middle] < value) {
            low = middle + 1;
        } else {
            high = middle - 1;
        }
    }

    throw new Error('Value not in array');
};

File: python/binary-search/binary_search.py
def find(search_list, value):

    low = 0
    high = len(search_list) - 1

    while low <= high:
        mid = (low + high) // 2
        if search_list[mid] < value:
            low = mid + 1
        elif search_list[mid] > value:
            high = mid - 1
        else:
            return mid

    raise ValueError("value not in array")
